packages/react-components/src/components/CheckBoxGroup.tsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import { useRuleEngine, useFormIntl } from '@aemforms/af-react-renderer';
+import {State, FieldJson} from '@aemforms/af-core';
+
+const CheckBoxGroup = (fieldjson: State<FieldJson>) => {
+   const [props, handlers] = useRuleEngine(fieldjson);
+   const { id, label, required, enumNames, enum: enums, value, name, valid, errorMessage, description } = props;
+   const i18n = useFormIntl();
+   const newVal = !value ? [] : value instanceof Array ? value : [value];
+   const options = enumNames && enumNames.length ? enumNames : enums || [];
+   const validateState = valid === false ? 'invalid' : 'valid';
+   const error = validateState === 'invalid';
+   const formattedDescription = i18n.formatMessage({ id: `${description}` });
+   const formattedLabel = i18n.formatMessage({ id: `${label?.value}` });
+  
+   const changeHandler = (event: React.ChangeEvent<HTMLInputElement>) => {
+      const val = event.target.value;
+      const checked = event.target.checked;
+      let valAdded = [...newVal];
+      if (checked) {
+        valAdded.push(val);
+      }
+      if (!checked) {
+        valAdded = valAdded.filter((item) => item !== val);
+      }
+      handlers.dispatchChange(valAdded);
+    };
+   return (
+        <div className="cmp-adaptiveform-checkboxgroup">
+            <label htmlFor={id} className="cmp-adaptiveform-checkboxgroup__label">{formattedLabel}</label>
+           <div className="cmp-adaptiveform-checkboxgroup__widget HORIZONTAL">
+              {options?.map((item: string, index: number) => {
+                return (
+                    <div className="cmp-adaptiveform-checkboxgroup-item [${id}]">
+                      <label className="cmp-adaptiveform-checkbox__label" aria-label={enumNames![index]} htmlFor={`${id}_${enumNames![index]}__widget`}>
```

Comment:
enumNames![index]?
`${id}_${enumNames![index]}__widget`?

Reply:
Sometimes the TypeScript compiler isn’t able to determine what type of value it may at a certain point.
By adding the exclamation mark (!) at the end, we let the TypeScript compiler that there is no way this variable will be undefined or null.